Wave watch support

fischer-ncar opened the pull request:
config_compsets.xml:  Turned on WaveWatch for B1850 and BHISTC5L40CNR compsets.  Other
                                     B1850s and BHISTs compsets have a Ws|d added to indicate which wave 
                                     model is used SWAV or DWAV.  WW3(WaveWatch) is the default.
testlist_allactive.xml:  Update compset aliases to match new compset names
config_pes.xml:   WaveWatch has a maximum number of 120 processors it can use.  So limit the 
                           number of processors for WaveWatch to 120.
config_grids.xml:  Add ww3a grid for WaveWatch model, make sure null grid is used when not using
                            WaveWatch

Test suite:  Ran namelist tests for prealpha and prebeta on yellowstone for intel, gnu, and pgi.
                   Manually checked to see if WaveWatch grid was set correctly in env_build.xml
Test baseline: 
Test namelist changes: B1850 and BHIST compset names changed by adding Ws|d to indicate
                                     if SWAV or DWAV is being used. 
Test status: B1850 and BHISTC5L40CNR will have answer differences since WaveWatch was
                    turned on for these compsets.

Fixes: [CIME Github issue #]

User interface changes?: 

Code review:

fischer-ncar committed:
Remove ESMF tests and add ChangeLog

fischer-ncar committed:
Updates for WaveWatch

mvertens commented:
We need to sit down with Jim tomorrow and iron out the aliases.

mvertens commented on cime_config/cesm/allactive/config_compsets.xml:
```diff
@@ -46,37 +46,42 @@
 
   <compset>
-    <alias>B1850RG</alias>
+    <alias>B1850RGWs</alias>
```
I think we need only 1 B1850 compset alias at this point and that is you B1850 compset. We do not need the other aliases.
Similarly I think we need only 1 BHIST that corresponds to the B1850.  I think we only need to test these.
From the point of view of the fully active compsets I would err on a very sparse set and let any deviation be a user defined compset. Before you implement these changes lets sit down and talk to Jim as well.

mvertens replied:
I'd like to understand the requirement for an out of the box WACCM compset with all active components.

fischer-ncar committed:
Removed allactive compsets that weren't being tested.

Test suite:
Test baseline:
Test namelist changes:
Test status: [bit for bit, roundoff, climate changing]

Fixes: [CIME Github issue #]

User interface changes?:

Code review:

fischer-ncar commented:
Removed 13 allactive compsets that weren't being tested.

jedwards4b commented:
Review complete - looks good.

fischer-ncar committed:
Update ChangeLog, fix typo in config_compset.xml

Test suite:
Test baseline:
Test namelist changes:
Test status: [bit for bit, roundoff, climate changing]

Fixes: [CIME Github issue #]

User interface changes?:

Code review:
